modules/internal/ChapelUtil.chpl

```diff
@@ -97,6 +97,21 @@ module ChapelUtil {
       }
     }
   }
+
+  //
+  // safeMul: If a and b are of type t, return true iff no
+  //  overflow/underflow would occur for a * b
+  //
+  proc safeMul(a: ?t, b:t){
+    if !isIntegralType(t) then
+      compilerError("Values must be of integral type.");
+    if a==-1 && b == min(t) then return false;
+    if b==-1 && a == min(t) then return false; 
+    if a > max(t)/b then return false;
+    if a < min(t)/b then return false;
+    //if all the case are false then multiplication is safe
+    return true;
+  }
```

Comment:
It is good to have this. But it is not a string-only helper so calls for some testing. See `test/modules/sungeun/test_safeAdd.chpl`. Let's try to write a similar test for `safeMul`.